### PR TITLE
Adding Virus Total Search Capabilities

### PR DESCRIPTION
**Pull request by secops-account**
Added search feature to VTLookUpV3 class....

Search is a premium feature and requires a Virus Total Enterprise Account.

VirusTotal Intelligence allows you to search through our dataset in order to identify files that match certain criteria (hash, antivirus detections, metadata, submission file names, file format structural properties, file size, etc.). We could say that it is pretty much like the "Google" of malware. (source: https://docs.virustotal.com/docs/virustotal-intelligence-introduction)

For additional documentation, and examples please see `docs > notebooks > VirusTotalSearch.ipynb`

### CODE REVIEW

**Comment by secops-account** [comment]
> @secops-account please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

File: msticpy/context/vtlookupv3/vtlookupv3.py [hunk 1]
```diff
@@ -386,7 +389,7 @@ async def _lookup_ioc_relationships_async(
         observable: str,
         vt_type: str,
         relationship: str,
-        limit: Optional[int] = None,
+        limit: int = None,
```
Comment (ianhelle) [inline msticpy/context/vtlookupv3/vtlookupv3.py:392]: Not clear why you're changing the type annotation here - if it's a nullable int it should be Optional[int]

File: msticpy/context/vtlookupv3/vtlookupv3.py [hunk 2]
```diff
@@ -506,9 +509,8 @@ def lookup_ioc_relationships(
         observable: str,
         vt_type: str,
         relationship: str,
-        limit: Optional[int] = None,
```
Comment (ianhelle) [inline msticpy/context/vtlookupv3/vtlookupv3.py:509]: Not clear why you're changing the type annotation here - if it's a nullable int it should be Optional[int]

File: msticpy/context/vtlookupv3/vtlookupv3.py [hunk 3]
```diff
@@ -687,7 +678,7 @@ def lookup_iocs_relationships(
         relationship: str,
         observable_column: str = ColumnNames.TARGET.value,
         observable_type_column: str = ColumnNames.TARGET_TYPE.value,
-        limit: Optional[int] = None,
+        limit: int = None,
```
Comment (ianhelle) [inline msticpy/context/vtlookupv3/vtlookupv3.py:681]: Not clear why you're changing the type annotation here - if it's a nullable int it should be Optional[int]

File: msticpy/context/vtlookupv3/vtlookupv3.py [hunk 4]
```diff
@@ -879,6 +870,101 @@ def get_file_behavior(
         vt_behavior.get_file_behavior(sandbox=sandbox)
         return vt_behavior
 
+
```
Comment (ianhelle) [inline msticpy/context/vtlookupv3/vtlookupv3.py:873]: Nice work!

**Review by ianhelle — CHANGES_REQUESTED** [review]
Overall looks great but I had questions about some of the changes.

File: msticpy/context/vtlookupv3/vtlookupv3.py (hunk 1 above)
Comment (secops-account) [reply msticpy/context/vtlookupv3/vtlookupv3.py]: Acknowledged - Please see changes [in commit:817c6607afd96b6dee427e10c9c36bc733a13b77](https://github.com/microsoft/msticpy/pull/739/commits/817c6607afd96b6dee427e10c9c36bc733a13b77)

File: msticpy/context/vtlookupv3/vtlookupv3.py (hunk 2 above)
Comment (secops-account) [reply msticpy/context/vtlookupv3/vtlookupv3.py:509]: Acknowledged - Please see changes [in commit:817c6607afd96b6dee427e10c9c36bc733a13b77](https://github.com/microsoft/msticpy/pull/739/commits/817c6607afd96b6dee427e10c9c36bc733a13b77)

File: msticpy/context/vtlookupv3/vtlookupv3.py (hunk 3 above)
Comment (secops-account) [reply msticpy/context/vtlookupv3/vtlookupv3.py]: Acknowledged - Please see changes [in commit:817c6607afd96b6dee427e10c9c36bc733a13b77](https://github.com/microsoft/msticpy/pull/739/commits/817c6607afd96b6dee427e10c9c36bc733a13b77)

**Review by ianhelle — APPROVED** [review]

**Pull request merged**